eo-maven-plugin/src/test/java/org/eolang/maven/testapi/CompilationResult.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2022 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.testapi;
+
+import com.yegor256.tojos.TjSmart;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.eolang.maven.Catalogs;
+import org.eolang.maven.Home;
+import org.eolang.maven.ParseMojo;
+import org.eolang.maven.TranspileMojo;
+
+/**
+ * Maven compilation result.
+ * @since 0.28.12
+ */
+public final class CompilationResult {
```

Comment:
@volodya-lombrozo it seems like we are doing here what Hamcrest matchers must do. I think it's better to stay with standard matchers for a while. Then, when we see that certain matchers are used very often, we will introduce our own. Making `FakeMaven` do the assertion is a mix of concepts in one place, I would say (a violation of SRP, if you wish).

Reply:
@yegor256 
Well... I can remove it, I see that it definitely looks like overdesign, at lest of now, but I tried to avoid violation of SRP exactly by using  `CompilationResult`  class. Also, I wanna hide complexity from programmers eyes. Just compare:

```java
  MatcherAssert.assertThat(
            new Home(target).exists(
                Paths.get(
                    String.format("%s/foo/x/main.%s", ParseMojo.DIR, TranspileMojo.EXT)
                )
            ),
            Matchers.is(true)
        );
        MatcherAssert.assertThat(
            new TjSmart(
                Catalogs.INSTANCE.make(foreign)
            ).getById("foo.x.main").exists("xmir"),
            Matchers.is(true)
        );
``` 

with: 
```java
MatcherAssert.assertThat(result, new XmirCreatedWell()));
```
looks beautiful, isn't?

Moreover, In some checks we have to know where `target` dir is , where `eo-foreign.json` is placed and etc. In other words, I need to know internal parts of `FakeMaven`. What should I do? (Write a getter? :)  By using `CompilationResult` I can encapsulate all required internals.